thorlcr/thorutil/thbuf.cpp

```diff
@@ -284,6 +284,8 @@ class CSmartRowBuffer: public CSimpleInterface, implements ISmartRowBuffer, impl
         while (out->ordinality()) 
             ReleaseThorRow(out->dequeue());
         delete out;
+        if (file)
```

Comment:
would this be better if it checked fileio, and cleared fileio before removing the file?  I suspect it works on linux, but may have problems on windows/other OS.

Also IFile::remove, it can throw exceptions - which you don't really want in a destructor.  (I'm not sure it is a good idea that it can throw exceptions....)